src/flake8_tidy_imports/__init__.py

```diff
@@ -54,6 +56,8 @@ def parse_options(cls, options: Any) -> None:
             line.strip() for line in options.banned_modules.split("\n") if line.strip()
         ]
         cls.banned_modules = {}
+        cls.banned_patterns = {}
```

Comment:
Possibly premature optimization - since we have to iterate through all of these for each import, I tried to keep this small. We still key for an exact match in `banned_modules` (as a dictionary lookup) first.